tests/test_types.py

```diff
@@ -79,6 +80,60 @@ def test_constrained_bytes_too_long():
     ]
 
 
+class ConListModelMax(BaseModel):
+    v: conlist(int, max_items=10) = []
+
+
+class ConListModelMin(BaseModel):
+    v: conlist(int, min_items=1) = []
+
+
+def test_constrained_list_good():
+    m = ConListModelMax(v=[1, 2, 3])
+    assert m.v == [1, 2, 3]
+
+
+def test_constrained_list_default():
+    m = ConListModelMax()
+    assert m.v == []
+
+
+def test_constrained_list_too_long():
+    with pytest.raises(ValidationError) as exc_info:
+        ConListModelMax(v=list(str(i) for i in range(11)))
+    assert exc_info.value.errors() == [
+        {
+            'loc': ('v',),
+            'msg': 'ensure this value has at most 10 items',
+            'type': 'value_error.list.max_items',
+            'ctx': {'limit_value': 10},
+        }
+    ]
+
+
+def test_constrained_list_too_short():
+    with pytest.raises(ValidationError) as exc_info:
+        ConListModelMin(v=[])
+    assert exc_info.value.errors() == [
+        {
+            'loc': ('v',),
+            'msg': 'ensure this value has at least 1 items',
+            'type': 'value_error.list.min_items',
+            'ctx': {'limit_value': 1},
+        }
+    ]
+
+
+def test_constrained_list_item_type_fails():
+    with pytest.raises(ValidationError) as exc_info:
+        ConListModelMax(v=['a', 'b', 'c'])
+    assert exc_info.value.errors() == [
+        {'loc': ('v', 0), 'msg': 'value is not a valid integer', 'type': 'type_error.integer'},
+        {'loc': ('v', 1), 'msg': 'value is not a valid integer', 'type': 'type_error.integer'},
+        {'loc': ('v', 2), 'msg': 'value is not a valid integer', 'type': 'type_error.integer'},
+    ]
```

Comment:
please add a test for a customised constrained list, something like

```py
class SevenElevanList(ConstrainedList):
    min_items = 7
    max_items = 11

class MyModel(BaseModel):
    foobar: SevenElevanList[float]

tests....
```

Reply:
Done